pilot/pkg/serviceregistry/kube/client.go

```diff
@@ -37,7 +39,7 @@ import (
 func ResolveConfig(kubeconfig string) (string, error) {
 	// Consistency with kubectl
 	if kubeconfig == "" {
-		kubeconfig = os.Getenv("KUBECONFIG")
+		kubeconfig = pilot.KubeConfig
```

Comment:
I think we should get rid of this function ?

Reply:
Indeed, this is only used by e2e.